drivers/spi/spi_litex_litespi.c

```diff
@@ -244,6 +285,21 @@ static int spi_litex_release(const struct device *dev, const struct spi_config *
 	return 0;
 }
 
+#if SPI_LITEX_ANY_HAS_IRQ
+static void spi_litex_irq_handler(const struct device *dev)
+{
+	struct spi_litex_data *data = dev->data;
+	const struct spi_litex_dev_config *dev_config = dev->config;
+
+	if (litex_read8(dev_config->core_master_ev_pending_addr) & BIT(0)) {
+		k_sem_give(&data->sem_rx_ready);
```

Comment:
You should handle Rx directly in isr mode, it will be much faster, no need of any smartphone then.